app/code/Magento/CustomerGraphQl/Model/Customer/UpdateCustomerData.php

```diff
@@ -19,6 +19,8 @@
  */
 class UpdateCustomerData
 {
+    private const RESTRICTED_DATA_KEYS = ['email', 'password'];
```

Comment:
Please move this data to constructor parameter (values should be set via di)
It more extensible way

BTW maybe we can reuse some data provider for these fields? Do we have the corresponding model in Magento?
Thanks